src/Components/Aspire.Confluent.Kafka/ConfluentKafkaMetrics.cs

```diff
@@ -0,0 +1,91 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Concurrent;
+using System.Diagnostics.Metrics;
+
+namespace Aspire.Confluent.Kafka;
+
+internal sealed class ConfluentKafkaMetrics
+{
+    private readonly IMeterFactory _meterFactory;
+    private readonly Meter _meter;
+    public Counter<long> Tx { get; }
+    public Counter<long> TxBytes { get; }
+    public Counter<long> TxMessages { get; }
+    public Counter<long> TxMessageBytes { get; }
+    public Counter<long> Rx { get; }
+    public Counter<long> RxBytes { get; }
+    public Counter<long> RxMessages { get; }
+    public Counter<long> RxMessageBytes { get; }
+
+    public ConcurrentQueue<Measurement<long>> ReplyQueueMeasurements { get; } = new ConcurrentQueue<Measurement<long>>();
+    public ConcurrentQueue<Measurement<long>> MessageCountMeasurements { get; } = new ConcurrentQueue<Measurement<long>>();
+    public ConcurrentQueue<Measurement<long>> MessageSizeMeasurements { get; } = new ConcurrentQueue<Measurement<long>>();
+
+    public ConfluentKafkaMetrics(IMeterFactory meterFactory)
+    {
+        _meterFactory = meterFactory;
+        _meter = _meterFactory.Create(ConfluentKafkaCommon.MeterName);
+
+        _meter.CreateObservableGauge(Counters.ReplyQueue, GetReplyQMeasurements);
+        _meter.CreateObservableGauge(Counters.MessageCount, GetMessageCountMeasurements);
+        _meter.CreateObservableGauge(Counters.MessageSize, GetMessageSizeMeasurements);
+
+        Tx = _meter.CreateCounter<long>(Counters.Tx);
+        TxBytes = _meter.CreateCounter<long>(Counters.TxBytes);
+        TxMessages = _meter.CreateCounter<long>(Counters.TxMessages);
+        TxMessageBytes = _meter.CreateCounter<long>(Counters.TxMessageBytes);
+        Rx = _meter.CreateCounter<long>(Counters.Rx);
+        RxBytes = _meter.CreateCounter<long>(Counters.RxBytes);
+        RxMessages = _meter.CreateCounter<long>(Counters.RxMessages);
+        RxMessageBytes = _meter.CreateCounter<long>(Counters.RxMessageBytes);
+    }
+
+    public static class Counters
+    {
+        private const string Prefix = "messaging.kafka.";
+        public const string ReplyQueue = Prefix + "replyq";
+        public const string MessageCount = Prefix + "msg_cnt";
+        public const string MessageSize = Prefix + "msg_size";
+        public const string Tx = Prefix + "tx";
+        public const string TxBytes = Prefix + "tx_bytes";
+        public const string Rx = Prefix + "rx";
+        public const string RxBytes = Prefix + "rx_bytes";
+        public const string TxMessages = Prefix + "txmsgs";
+        public const string TxMessageBytes = Prefix + "txmsg_bytes";
+        public const string RxMessages = Prefix + "rxmsgs";
+        public const string RxMessageBytes = Prefix + "rxmsg_bytes";
+    }
+
+    public static class Tags
+    {
+        public const string ClientId = "client_id";
```

Comment:
otel attribute is [`messaging.client_id`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/messaging.md). 

Also, it's not common to report client Id in the metrics because of potentially high cardinality (if user app creates a new instance every time their app is not only slow, but also now kafka metrics are capped and not useful). 
Instead of adding it as an attribute, could you just add a log whenever new client is created or have a counter for number of active client instances?

Reply:
@lmolkova  Updated the PR and renamed client_id attribute to messaging.client_id.